cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowNode.java

```diff
@@ -16,16 +16,24 @@
 
 package co.cask.cdap.api.workflow;
 
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import javax.annotation.Nullable;
+
 /**
  * Represents node in the {@link Workflow}.
  */
 public abstract class WorkflowNode {
   protected final String nodeId;
   protected final WorkflowNodeType nodeType;
+  protected final Set<String> parentNodeIds;
 
-  public WorkflowNode(String nodeId, WorkflowNodeType nodeType) {
+  public WorkflowNode(String nodeId, WorkflowNodeType nodeType, @Nullable Set<String> parentNodeIds) {
```

Comment:
When we configure the Workflow, we do not add the parent ids and they are null. Parent id's are added during generating the specification for the Workflow.